etna/models/nn/deepstate/state_space_model.py

```diff
@@ -0,0 +1,445 @@
+
+        Parameters
+        ----------
+        datetime_index:
+            Tensor with the index values.
+            Values should be from 0 to seasonal period.
+
+        Returns
+        -------
+        :
+            Emission coefficient matrix.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def transition_coeff(self, datetime_index: Tensor) -> Tensor:  # (latent_dim, latent_dim)
+        """Transition coefficient matrix.
+
+        Parameters
+        ----------
+        datetime_index:
+            Tensor with the index values.
+            Values should be from 0 to seasonal period.
+
+        Returns
+        -------
+        :
+            Transition coefficient matrix.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def innovation_coeff(self, datetime_index: Tensor) -> Tensor:  # (batch_size, seq_length, latent_dim)
+        """Innovation coefficient matrix.
+
+        Parameters
+        ----------
+        datetime_index:
+            Tensor with the index values.
+            Values should be from 0 to seasonal period.
+
+        Returns
+        -------
+        :
+            Innovation coefficient matrix.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def generate_datetime_index(self, timestamps: np.ndarray) -> np.ndarray:
+        """Generate datetime index to use in the State Space Model.
+
+        Parameters
+        ----------
+        timestamps:
+            Array with timestamps.
+
+        Returns
+        -------
+        :
+            Datetime index for State Space Model.
+        """
+        raise NotImplementedError
+
+
+class LevelSSM(SSM):
+    """Class for Level State Space Model."""
+
+    def latent_dim(self) -> int:
+        """Dimension of the latent space.
+
+        Returns
+        -------
+        :
+            Dimension of the latent space.
+        """
+        return 1
+
+    def emission_coeff(self, datetime_index: Tensor) -> Tensor:
+        """Emission coefficient matrix.
+
+        Parameters
+        ----------
+        datetime_index:
+            Tensor with the index values.
+            Values should be from 0 to seasonal period.
+
+        Returns
+        -------
+        :
+            Emission coefficient matrix.
+        """
+        batch_size, seq_length = datetime_index.shape[0], datetime_index.shape[1]
+        emission_coeff = torch.ones(batch_size, seq_length, self.latent_dim(), device=datetime_index.device)
+        return emission_coeff.float()
+
+    def transition_coeff(self, datetime_index: Tensor) -> Tensor:
+        """Transition coefficient matrix.
+
+        Parameters
+        ----------
+        datetime_index:
+            Tensor with the index values.
+            Values should be from 0 to seasonal period.
+
+        Returns
+        -------
+        :
+            Transition coefficient matrix.
+        """
+        transition_coeff = torch.eye(self.latent_dim())
```

Comment:
Don't we need to set the device here as well -- I guess this should solve the problems in deepstate